cmd/libs/go2idl/client-gen/main.go

```diff
@@ -39,21 +40,33 @@ var (
 	fakeClient    = flag.Bool("fake-clientset", true, "when set, client-gen will generate the fake clientset that can be used in tests")
 )
 
-func versionToPath(group string, version string) (path string) {
+func versionToPath(gvPath string, group string, version string) (path string) {
 	const base = "k8s.io/kubernetes/pkg"
 	// special case for the core group
 	if group == "api" {
 		path = filepath.Join(base, "api", version)
 	} else {
-		path = filepath.Join(base, "apis", group, version)
+		path = filepath.Join(base, "apis", gvPath, group, version)
 	}
 	return
 }
 
+func parsePathGroupVersion(pgvString string) (gvPath string, gvString string) {
```

Comment:
On second thought, this is how I envision a user will use client-gen:
```
./client-gen --input="pkg/federations/apis/fedatration/, pkg/apis/extensions/, pkg/api/" --clientset-path=... --clientset-name=...
```
And client-gen would parse the group and version from the full path. What do you think?